### ISSUE TITLE
Radio inputs are not grouped and labeled

### ISSUE THREAD

**Issue by steverep** [home-assistant/frontend#14146]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

All instances of `ha-radio` and `mwc-radio` are not properly marked up for accessibility.  They need to not only be associated with their individual labels, but also grouped and labeled with the associated preceding question.  Without this, filling out the form with a screen reader is more difficult since the question is not read when tabbing to the radios.  Also, keyboard focus should be managed such that there is only one tab stop and arrow keys can be used to make the selection.

_Note: This also applies to checkbox groups, except for the keyboard part._

### Describe the behavior you expected

All radio inputs should follow the [ARIA radio group pattern](https://www.w3.org/WAI/ARIA/apg/patterns/radiobutton/).

### Steps to reproduce the issue

n/a

### What version of Home Assistant Core has the issue?

2022.10.x

### In which browser are you experiencing the issue with?

n/a

### Which operating system are you using to run this browser?

n/a

**Comment by steverep** [comment]
Still an issue

**Comment by KTibow** [comment]
Could this possibly be resolved by changing ha-formfield to wrap the whole thing in a label instead of just the text?

**Comment by steverep** [comment]
> Could this possibly be resolved by changing ha-formfield to wrap the whole thing in a label instead of just the text?

No, `ha-form-field` only labels a single radio input.  Grouping requires using native `fieldset`, which gives native keyboard control but doesn't necessarily play well with all user agents when shadow roots are involved, or ARIA., where keyboard control must be provided manually.

**Comment by bramkragten** [comment]
Is this solved in `md-radio`?

**Comment by silamon** [comment]
> Is this solved in `md-radio`?

I can confirm that it is solvable with the new material design components although it would still be a little bit refactoring.

**Comment by steverep** [comment]
> > Is this solved in `md-radio`?
> 
> I can confirm that it is solvable with the new material design components although it would still be a little bit refactoring.


Sorry @bramkragten your question fell through the cracks for me.  Based on their docs I'd say no, but after playing with their demo (which didn't exist last I checked) and then digging into their code, they do seem to handle keyboard function properly.  It looks like they do it with a reactive controller inside the radio component to implement a rolling `tabindex` pattern on siblings of the host:

https://github.com/material-components/material-web/blob/c97362c1ccb30e7bd19316986ee91a767ce02423/radio/internal/single-selection-controller.ts#L10

Their docs don't mention this at all or any potential restrictions it might impose, so I'd be cautious at first.  I will look into it more.
